webApp/src/main/java/com/infoshare/controller/NeedRequestController.java

```diff
@@ -58,6 +63,14 @@ public NeedRequestController(NeedRequestService needRequestService, NeedRequestF
     @PostMapping("/filtering")
     public String filtering(@ModelAttribute(FILTER_FORM_ATTR) NeedRequestFilterForm form,
                             RedirectAttributes redirectAttributes) {
+
+        final String uri = "http://localhost:8081/needRequestSearchStats";
+        RestTemplate restTemplate = new RestTemplate();
```

Comment:
Ten rodzaj kodu jest swietnym kandydatem do uzycia aspektow w Springu: https://nullpointerexception.pl/jak-korzystac-ze-spring-aop/ . Jesli nie chce Ci sie przerobic tego na askpety, to wydziel ten kod do oddzielnego serwisu  i zrob stad jedynie wywolanie tej metody.